icefoganalytics/internal-data-portal
# Data Owner: Ability To Add a New Dataset

Fixes https://github.com/icefoganalytics/internal-data-portal/issues/16

Relates to:
- [WireFrames, Internal Data Portal, 2024-01-08.pdf](https://icefoganalyticscom.sharepoint.com/:b:/s/Project_Files/EQ6oYyRWcnNNqYFUX7qn2H4B48jY4Jsm1zNt6OHUaIVkIA?e=Lz88mF)
- [User Stories, Internal Data Portal, 2024-01-08.docx](https://icefoganalyticscom.sharepoint.com/:w:/r/sites/Project_Files/_layouts/15/Doc.aspx?sourcedoc=%7BAA0A40A5-3CE8-4021-9468-C60ADCE1DAF4%7D&file=UseCases.docx&action=default&mobileredirect=true)
- [0001, Data Modeling, 2024-01-05](https://docs.google.com/document/d/1Mdbl_Cy0BgKiq0DGdvOKUqJn-FdgbF0LSVykwf__FY8/edit?pli=1)
- [0002, UI Modeling, 2024-01-08](https://docs.google.com/document/d/1tP63Ob4XPBGlASPHVf1GnNb3hlIkybmWBqJDA61u3J0/edit)

# Context

Build UI for Data Owner to create a new dataset see [0002, UI Modeling, 2024-01-08](https://docs.google.com/document/d/1tP63Ob4XPBGlASPHVf1GnNb3hlIkybmWBqJDA61u3J0/edit#heading=h.gxfmpcvkuit1)

I'll need to have a URL for dataset, which is currently not shown in the wireframes.
- `/datasets/new` or something, instead of the edit page located at `/datasets/:slug/description/manage`

The /new page will have the minimal fields required to create the dataset and have no tabs? Alternatively, it could have the tabs, but have them greyed out. Either way, the /new page will be as minimal as possible, leaving the full feature set to the /manage page

![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/56f1f8ac-ddb2-48b9-9d20-7f4cd4a4a161)


# Implementation

Add a page to create new datasets. This page is not fully built out as the department/division/branch/unit data is fake.

# Screenshots

http://localhost:8080/datasets/new
![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/b8ec9a88-9ed7-4b65-bb9d-55f7899cf8f4)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. After being redirected to http://localhost:8080/dashboard, click the "create dataset" button.
5. Fill in the various fields to unlock the save button and save the dataset.
> Note that the organizational hierarchy data is fake at this time.
> Also note that only the current user can be selected.
